File: competition_upsolve/ioi10_cluedo/cluedo.pas
unit cluedo;

interface
   procedure Solve;
implementation
   uses graderlib;
   procedure Solve;
      var r: longint;
   begin
      r := Theory(1,2,3);
      if r <> 0 then r := Theory(3,2,1);
      if r <> 0 then r := Theory(4,4,4);
   end;
begin
end.

File: competition_upsolve/ioi10_cluedo/cluedo.c
#include "grader.h"
#include "cluedo.h"

void Solve(){
   int r;
   r = Theory(1,2,3);
   if (r == 0) return;
   r = Theory(3,2,1);
   if (r == 0) return;
   r = Theory(4,4,4);
   if (r == 0) return;
}
